### PR TITLE
Concept Edit: JavaScript Dynamic Typing

### PR DESCRIPTION
**Pull request by GregV3**
### Description

I've decided to update the text about JavaScript's variables in the location Docs/JavaScript/Variables. I think it is worth mentioning that it is one of the dynamic typing languages.

### Type of Change

- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@GregV3, I did a first review and left a couple of comments. You also need to sign the [CLA](https://cla-assistant.io/Codecademy/docs?pullRequest=1344).

File: content/javascript/concepts/variables/variables.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 Title: 'Variables'
-Description: 'Variables are used whenever there’s a need to store a piece of data. A variable contains data that can be used in the program elsewhere. Using variables also ensures code re-usability since it can be used to replace the same value in multiple places. js const currency = $; let userIncome = 85000; console.log(currency + userIncome +  is more than the average income.); // Output: $85000 is more than the average income. '
+Description: The text about assigning variables and dynamic typing in JavaScript. 
 Subjects:
   - 'Web Development'
   - 'Computer Science'
 Tags:
   - 'Variables'
   - 'Data Types'
+  - 'Dynamic typing'
```
[the remark this answers is not in the document]
Comment (yangc95) [reply content/javascript/concepts/variables/variables.md:10]:
```suggestion
```

**Comment by yangc95** [comment]
@GregV3 @SSwiniarski I helped push up some of the requested changes! Is there anything else that needs improving before moving this PR on to a second review?

TY for everyone's efforts here ^^ this looks to be a short and sweet addition to this entry.

**Review by SSwiniarski — APPROVED** [review]
@yangc95, @GregV3 Looks good to merge.

**Review by yangc95 — APPROVED** [review]

File: content/javascript/concepts/variables/variables.md [hunk 2]
```diff
@@ -35,3 +35,12 @@ var age;
 let weight;
 const numberOfFingers = 20;
 ```
+
+## Dynamic typing
```
Comment (yangc95) [inline content/javascript/concepts/variables/variables.md:39]:
```suggestion
## Dynamic Typing
```

File: content/javascript/concepts/variables/variables.md [hunk 3]
```diff
@@ -35,3 +35,12 @@ var age;
...
+
+Unlike many programming languages, JavaScript doesn't force the user to specify a variable's data type. It assumes the data type based on the way a user assigns a value. For example, the example below assigns a string to a variable by putting the text between quotation marks. Without them, JavaScript reads it as a number. This is why it's considered to be a dynamically-typed language.
```
Comment (yangc95) [inline content/javascript/concepts/variables/variables.md:41]:
```suggestion
Unlike many programming languages, JavaScript doesn't force the user to specify a variable's [data type](https://www.codecademy.com/resources/docs/javascript/data-types). It assumes the data type based on the value. The example below assigns a string to a variable by enclosing some numbers in single quotation marks. Without them, JavaScript reads the value as a number.
```

File: content/javascript/concepts/variables/variables.md [hunk 4]
```diff
@@ -35,3 +35,12 @@ var age;
 let weight;
 const numberOfFingers = 20;
 ```
+
+## Dynamic Typing
+
+Unlike many programming languages, JavaScript doesn't force the user to specify a variable's [data type](https://www.codecademy.com/resources/docs/javascript/data-types). It assumes the data type based on the value. The example below assigns a string to a variable by enclosing some numbers in single quotation marks. Without them, JavaScript reads the value as a number.
```
Comment (yangc95) [inline content/javascript/concepts/variables/variables.md:41]:
```suggestion
Unlike many programming languages, JavaScript doesn't require the user to specify a variable's [data type](https://www.codecademy.com/resources/docs/javascript/data-types). It assumes the data type based on the value. The example below assigns a string to a variable by enclosing some numbers in single quotation marks. Without them, JavaScript reads the value as a number.
```

**Pull request merged**